### ISSUE TITLE
Problem when running Emacs as systemd user unit

### ISSUE THREAD

**Issue by andreas-h** [tarsius/moody#4]
When I start Emacs via systemd (i.e., on Linux) and then connect via emacsclient, moody's "graphical" elements (i.e., the diagonals) are broken.  Instead, I see empty boxes:

![bildschirmfoto_2018-03-05_14-15-15](https://user-images.githubusercontent.com/358378/36976948-a9daabaa-207f-11e8-9abc-f89d17665375.png)

When I start Emacs "normally", everything looks as expected (also when connecting using emacsclient to a server process that was *not* started via systemd).

Any ideas what might be the problem here?

**Comment by tarsius** [comment]
> Any ideas what might be the problem here?

Afraid not.

**Comment by andreas-h** [comment]
Mhh, seems there are several problems here.

Only the frame which is in focus is affected, the other frames look alright:

![bildschirmfoto_2018-03-05_15-24-12](https://user-images.githubusercontent.com/358378/36980441-8be4708a-208a-11e8-8d7f-678deff0cb41.png)

Unfortunately I don't have another setup here at home; when I'm back to work I'll try out if the same happens on Cinnamon/Gnome.

When I *don't* use systemd and open several frames on the same desktop, in some frames I have a similar problem to #4. As soon as I change focus to any Emacs frame on that window, the problem is gone. Unfortunately I cannot give a screenshot for this one, since it seems that Xfce4 does make a focus shift when I take the screenshot, leading to the problem being fixed at the time when the screenshot is actually taken.

**Comment by andreas-h** [comment]
Okay, problem is only in the active *window*:  When I have one frame with two windows, only the active window shows the problem, while the other window looks fine.

**Comment by andreas-h** [comment]
Okay, I could pin the issue down to smart-mode-line.  As long as I don't  `(use-package smart-mode-line :config (sml/setup))`, everything (except for the not-loaded sml) works.

**Comment by tarsius** [comment]
I used `smart-mode-line` before I wrote this and `minions`. For now I am not using it anymore but I have reported a few issues and made some requests in its issue tracker yesterday. I think it is better to not use `sml/setup` when also using `moody` and to just add individual elements from that package. Compatibility will probably improve over time.

I don't think `moody` is at fault here, but I also have a hard time imagining how `smart-mode-line` could be at fault. If you report this in the `smart-mode-line` issue tracker, then first really make sure this problem only occurs when using systemd. (And if that is so, then I doubt that one of these package could be to blame grows even more.) You might also want to search the `powerline` issue tracker to see if a similar issue was reported there.
